engine/kv_engine.cpp

```diff
@@ -873,14 +873,18 @@ Status KVEngine::HashGetImpl(const StringView& key, std::string* value,
     if (record_size > configs_.pmem_segment_blocks * configs_.pmem_block_size) {
       continue;
     }
-    char data_buffer[record_size];
+
+    void* data_buffer = malloc(record_size);
```

Comment:
I'd rather not use malloc/free here. What do you think use std::string here for data_buffer?

Reply:
Actually malloc is okay, thread_local string is okay.
But we no longer need validation.